sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3848,6 +3848,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_IDENTIFIER_OUTPUT_CATALOG_NAME =
+    buildConf("spark.sql.legacy.identifierOutputCatalogName")
+      .internal()
+      .doc("When set to true, the identifier will output catalog name if database is defined. " +
+        "When set to false, it restores the legacy behavior that does not output catalog name.")
```

Comment:
nit: usually true means legacy behavior, we probably need to rename the config a little bit.